ai-service/src/ai_service/project_ingestor.py

```
@@ -0,0 +1,87 @@
+    Clones a GitHub repo to a temporary directory.
+    Returns the path to the cloned directory.
+    Validates the repo_url to prevent command injection.
+    """
+    # Only allow URLs matching the GitHub repo pattern
+    github_repo_pattern = r"^https://github\.com/[\w\-\.]+/[\w\-\.]+(\.git)?/?$"
+    if not re.match(github_repo_pattern, repo_url):
+        raise errors.InvalidParam.invalid_repo_url()
+    clone_to = tempfile.mkdtemp()
+    try:
+        Repo.clone_from(repo_url, clone_to)
+        return clone_to
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)